sympy/series/sequences.py

```diff
@@ -711,6 +713,160 @@ def coeff_mul(self, coeff):
         return SeqFormula(formula, self.args[1])
 
 
+class RecursiveSeq(SeqBase):
+
+    """A finite degree recursive sequence.
+
+    That is, a sequence a(n) that depends on a fixed, finite number of its
+    previous values. The general form is
+
+        a(n) = f(a(n - 1), a(n - 2), ..., a(n - d))
+
+    for some fixed, positive integer d, where f is some function defined by a
+    SymPy expression.
+
+    Parameters
+    ==========
+
+    initial : iterable with length equal to the degree of the recurrence
+        The initial values of the recurrence.
+
+    y : function with argument
+        The name of the recursively defined sequence, e.g., :code:`y(n)`.
+
+    recurrence : SymPy expression defining recurrence
+        This is *not* an equality, only the expression that the nth term is
+        equal to. For example, if :code:`a(n) = f(a(n - 1), ..., a(n - d))`,
+        then the expression should be :code:`f(a(n - 1), ..., a(n - d))`.
+
+    start : start value of sequence (inclusive)
+
+    stop : stop value of sequence (inclusive)
+
+    Examples
+    ========
+
+    >>> from sympy import Function, symbols
+    >>> from sympy.series.sequences import RecursiveSeq
+
+    >>> y = Function("y")
+    >>> n = symbols("n")
+    >>> fib = RecursiveSeq((0, 1), y(n), y(n - 1) + y(n - 2))
+
+    For value at a particular point
```

Comment:
Perhaps, this line (and similar) could be written as comments with the doctests?